BG-24608 Accept API Key in non-BitGo Recovery and Unsigned Sweep

glowang opened the pull request:
For non-BitGo recoveries (non-BitGo and unsigned sweep), we add a new field on the UI to accept a blockchair API key for bitcoin Mainnet and testnet recoveries.

I just tested that this worked for a btc prod wallet.

This change depends on the sdk change here: [https://github.com/BitGo/BitGoJS/pull/868](url)

glowang commented on src/components/non-bitgo.js:
```diff
@@ -149,6 +153,7 @@ class NonBitGoRecoveryForm extends Component {
 
       this.setState({ recovering: false, done: true, finalFilename: filePath });
     } catch (e) {
+      console.log(e)
```
This makes it easier to see the stack traces in console!

dpc commented on src/components/non-bitgo.js:
```diff
@@ -117,6 +117,10 @@ class NonBitGoRecoveryForm extends Component {
         return obj;
       }, {});
 
+      if(this.state.coin === 'btc'&& this.state.apiKey) {
```
Sorry about bothering about whitespaces, but at least in 4 places the formatting is off.

dpc replied:
And colons are missing on some lines.

dpc commented on src/components/unsigned-sweep.js:
```diff
@@ -176,15 +180,17 @@ class UnsignedSweep extends Component {
+      this.setState({ recovering: false, done: true, finalFilename: filePath.filePath });
+      alert('We recommend that you use a third-party API to decode your txHex and verify its accuracy before broadcasting.');
     } catch (e) {
+      console.dir(e);
```
Leftover or is this how we do logging in wrw? Just double checking.

glowang replied:
This logs the error to the console if there are any errors (which we would ask the customers to take a screenshot for)...There really isn't a standard practice as for how we log errors. Happy to hear if you have any other ideas on how we can do this!

dpc replied:
I would just wrap it in a `function log(s) { console.log(s) }` or something. Than it's self-explanatory for the reader that it's intentional. The function can be in some `src/log.js` file.

glowang replied:
Good idea, fixed!

twtaylorbitgo commented on src/constants/tooltips.js:
```diff
@@ -50,5 +57,10 @@ export default {
     scan: 'The amount of addresses without transactions to scan before stopping the tool.',
     tokenAddress: 'The address of the smart contract of the token to recover. This is unique to each token, and is NOT your wallet address.',
+    apiKey: (coin) => {
+      if( coin === 'btc') {
```
more spacing, please add a linter

glowang replied:
Fixed the spacing error, thanks for catching it! Yes we should add a linter, but I would prefer doing it outside the scope of this task. https://bitgoinc.atlassian.net/browse/BG-24127

glowang commented on src/components/unsigned-sweep.js:
```diff
@@ -176,15 +180,17 @@ class UnsignedSweep extends Component {
+      this.setState({ recovering: false, done: true, finalFilename: filePath.filePath });
+      alert('We recommend that you use a third-party API to decode your txHex and verify its accuracy before broadcasting.');
     } catch (e) {
+      console.dir(e);
```
This logs the error to the console if there are any errors (which we would ask the customers to take a screenshot for)... otherwise it would be kinda hard to see the stack traces when debugging. I see this `console.dir` as the easiest way to add logging with our time constraint.

glowang commented:
@dpc @twtaylorbitgo  thank you for reviewing, I addressed your feedback. The spacing errors and style issues will be addressed better via a linter in the future https://bitgoinc.atlassian.net/browse/BG-24127

twtaylorbitgo commented on src/components/non-bitgo.js:
```diff
@@ -139,18 +143,20 @@ class NonBitGoRecoveryForm extends Component {
-    }
+      this.setState({ recovering: false, done: true, finalFilename: [filePath.filePath] });
+      alert('We recommend that you use a third-party API to decode your txHex and verify its accuracy before broadcasting.');
+      } catch (e) {
+        logToConsole(e);
+        this.setState({ error: e.message, recovering: false });
+      }
```
this is still pretty bad from a formatting standpoint. please add a lint file

twtaylorbitgo commented on src/constants/tooltips.js:
```diff
@@ -50,5 +57,10 @@ export default {
     scan: 'The amount of addresses without transactions to scan before stopping the tool.',
     tokenAddress: 'The address of the smart contract of the token to recover. This is unique to each token, and is NOT your wallet address.',
+    apiKey: (coin) => {
+      if(coin === 'btc') {
```
more linting

twtaylorbitgo commented on src/components/non-bitgo.js:
```diff
@@ -117,6 +117,10 @@ class NonBitGoRecoveryForm extends Component {
         return obj;
       }, {});
 
+      if(this.state.coin === 'btc' && this.state.apiKey) {
```
...

twtaylorbitgo commented on src/components/non-bitgo.js:
```diff
@@ -139,16 +143,19 @@ class NonBitGoRecoveryForm extends Component {
 
-      this.setState({ recovering: false, done: true, finalFilename: filePath });
+      this.setState({ recovering: false, done: true, finalFilename: [filePath.filePath] });
+      alert('We recommend that you use a third-party API to decode your txHex a' +
+            'nd verify its accuracy before broadcasting.');
```
can we just make the 'and' a whole word on the nextline?

glowang replied:
we needed to break the sentence because it is over 80 char long

ericcrosson-bitgo replied:
We can use something like 

```
[
`We recommend that you use a third-party API to decode`
`your txHex and verify its accuracy before broadcasting`
].join(''
```

argjv commented on src/components/unsigned-sweep.js:
```diff
@@ -162,6 +162,10 @@ class UnsignedSweep extends Component {
         return obj;
       }, {});
 
+      if (this.state.coin === 'btc'&& this.state.apiKey) {
```
```suggestion
      if (this.state.coin === 'btc' && this.state.apiKey) {
```

argjv commented on src/constants/tooltips.js:
```diff
@@ -28,7 +28,14 @@ export default {
     scan: 'The amount of addresses without transactions to scan before stopping the tool.',
-    apiKey: 'An Api-Key Token from etherscan.com required for Ethereum Mainnet recoveries',
+    apiKey: (coin) => {
+      if (coin === 'eth' || coin === 'token') {
+        return 'An Api-Key Token from etherscan.com required for Ethereum Mainnet recoveries';
+      }
+      if (coin === 'btc') {
+        return 'An Api-Key Token from blockchair.com required for Bitcoin Mainnet and Testnet recoveries';
+      }
```
```suggestion
      if (coin === 'eth' || coin === 'token') {
        return 'An Api-Key Token from etherscan.com required for Ethereum Mainnet recoveries';
      } else if (coin === 'btc') {
        return 'An Api-Key Token from blockchair.com required for Bitcoin Mainnet and Testnet recoveries';
      } else {
        return 'An Api-Key Token required to fetch information from the external service and perform recoveries';
      }
```
